org.opent2t.sample.hub.superpopular/com.wink.hub/js/thingTranslator.js

```diff
@@ -88,6 +88,152 @@ class Translator {
+     * - Http server responds to the GET with the verificationResponseContent that was populated by subscribe
+     * - Future POST calls to the callback url are passed to the device translator get*Uri function
+     * 
+     * All calls to subscribe will return the expiration time of the subscription.  For Wink, this is
+     * currently 24 hours.  Subsequent calls to subscribe will refresh the expiration time, and in the
+     * case of Wink/Pubsubhubbub will not require verification.
+     * 
+     * @param {string} deviceType - Device Type (e.g. 'thermostats')
+     * @param {string|number} deviceId - Id for the specific device
+     * @param {string} callbackUrl - Callback url for feed postbacks
+     * @param {HttpRequest} verificationRequest - GET request received by the server at a previously subscribed
+     *      callback URL.  This completes the verification half of the subscription.
+     * @param {HttpResponse} verificationResponseContent - Content that should be provided in the response to the
+     *      verification request as {'Content-Type': 'text/plain'}.  All responses should use code 200 unless an
+     *      error is caught.
+     * @returns {number} Object containing the subscription expiration time, and any content that
+     *      needs to be included in a response to the original request.
+     */
+    _subscribe(deviceType, deviceId, callbackUrl, verificationRequest) {
+
+        if (callbackUrl) {
+            // Subscribe to notifications
+
+            var requestPath = '/' + deviceType + '/' + deviceId + '/subscriptions';
+
+            // Winks implementation of PubSubHubbub differs from the standard in that we do not need to provide
+            // the topic, or mode on this request.  Topic is implicit from the URL (deviceType/deviceId), and
+            // separate requests exist for mode (subscribe and unsubscribe vis POST/DELETE).
+
+            // Additionally, subscriptions will expire after 24 hours (for now), and need to be refreshed
+            // with another call to this function.
+
+            var postPayload = {
+                callback: callbackUrl,
+            }
+
+            var postPayloadString = JSON.stringify(postPayload);
+
+            return this._makeRequest(requestPath, 'POST', postPayloadString).then((response) => {
+                // Return the expiration time for the subscription
+                return {
+                    expiration: response.data.expires_at,
+                    response: ""
+                };
+            })
+
+        } else if (verificationRequest) {
+            // Verify a subscription request by constructing the appropriate response
+
+            var params = require('url').parse(verificationRequest.url, true, true);
+            // This is a subscription validation, populate the response
+            switch(params.query['hub.mode']) {
+                case "denied":
```

Comment:
>denied [](start = 22, length = 6)

instead of strings, are there error codes or HTTP status codes that we could look at?